File: blockchain/blockchain.py
import hashlib
import time
from datetime import datetime

block = 1
number = 1
previous = "0000000000000000000000000000000000000000000000000000000000000000"
text = "das ist ein test"

hashStr = f'{block}{number}{previous}{text}'
hashStr = hashStr.encode('utf-8')
hashStr = hashlib.sha256(hashStr)
now = datetime.now()
unix_start = time.time()
print("Start time: " + now.strftime("%d/%m/%Y %H:%M:%S"))
print("-----------------------------------------------------")

while hashStr.hexdigest()[:5] != "00000":
    number = number + 1
    hashStr = f'{block}{number}{previous}{text}'
    hashStr = hashStr.encode('utf-8')
    hashStr = hashlib.sha256(hashStr)

now = datetime.now()
unix_1 = time.time()
print("First block: " + now.strftime("%d/%m/%Y %H:%M:%S"))
print("First block: " + str((number / (unix_1 - unix_start))) + " hashes per second")
print(number)
print(hashStr.hexdigest())
print("-----------------------------------------------------")

block = 2
number = 1
previous = hashStr.hexdigest()
text = "das ist noch einmal ein test"

hashStr = f'{block}{number}{previous}{text}'
hashStr = hashStr.encode('utf-8')
hashStr = hashlib.sha256(hashStr)

while hashStr.hexdigest()[:5] != "00000":
    number = number + 1
    hashStr = f'{block}{number}{previous}{text}'
    hashStr = hashStr.encode('utf-8')
    hashStr = hashlib.sha256(hashStr)

now = datetime.now()
unix_2 = time.time()
print("Second block: " + now.strftime("%d/%m/%Y %H:%M:%S"))
print("First block: " + str((number / (unix_2 - unix_1))) + " hashes per second")
print(number)
print(hashStr.hexdigest())
print("-----------------------------------------------------")

File: blockchain/blockchain.js
const crypto = require('crypto');
let hashprefix = "00000";

//first block
let block = 1
let number = 1;
let previous = "0000000000000000000000000000000000000000000000000000000000000000"
let text = "das ist ein test"


let hash = crypto.createHash('sha256').update(block + number + previous + text).digest('hex');
console.log(new Date().toISOString());
while (hash.substring(0, hashprefix.length) != hashprefix) {
    number++;
    //console.log(number);
    hash = crypto.createHash('sha256').update(number + previous + text).digest('hex');
}

console.log(number + " " + hash);


//second block
block = 2
number = 1;
previous = hash
hash = ""
text = "das ist noch einmal ein test"
console.log(new Date().toISOString());
while (hash.substring(0, hashprefix.length) != hashprefix) {
    number++;
    //console.log(number);
    hash = crypto.createHash('sha256').update(number + previous + text).digest('hex');
}

console.log(number + " " + hash);
console.log(new Date().toISOString());